File: twopointer/011-container-with-most-water/ContainerWithMostWater.java
class Solution {
    public int maxArea(int[] height) {
        int left =0, right = height.length-1;
        int max =0;
        while(left<right){
            int min = Math.min(height[left], height[right]);
            max = Math.max((right-left)*min, max);
            while(left<right && height[left]<=min){
                left++;
            }
             while(left<right && height[right]<=min){
                right--;
            }
        }
        return max;
    }
}

File: twopointer/011-container-with-most-water/ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height)-1
        maxval = 0
        while left < right:
            minval = min(height[left],height[right])
            maxval = max((right-left)*minval, maxval)
            while(left<right and height[left]<= minval):
                left+=1
            while(left<right and height[right]<=minval):
                right-=1
        return maxval